### PR TITLE
feat: CI CD

### LINKED ISSUE #140: implement release workflow
npm? localhost? ipfs is no longer supported

_Originally posted by @mpetrunic in https://github.com/ChainSafe/metamask-snap-polkadot/pull/135#discussion_r920134002_

TODO:
implement ci workflow
fix default origin to point on npm package

Resources:
* https://engineering-handbook.onrender.com/development/tech-stack/typescript#releasing-npm-packages
* https://github.com/ChainSafe/filsnap

### PR DESCRIPTION
**Pull request by BeroBurny**
closes #140

we need env variables for `GITHUB_TOKEN` and `NPM_TOKEN`
cc @mpetrunic

### CODE REVIEW

File: .github/workflows/cd.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+name: Release
+on:
+  push:
+    branches:
+      - master
+jobs:
+  maybe-release:
```
Comment (mpetrunic) [inline .github/workflows/cd.yaml:7]: maybe better for this to depend on ci-workflow

File: .github/workflows/pr.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    steps:
+      - uses: amannn/action-semantic-pull-request@v4
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          types: |
+            fix
+            feat
+            chore
+          validateSingleCommit: true #single commit can ovveride squash merge commit message
```
Comment (mpetrunic) [inline .github/workflows/pr.yaml:21]: think you can remove this now, but make sure repo is following instructions in handbook

File: .github/.dependabot.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+version: 2
```
Comment (mpetrunic) [inline .github/.dependabot.yaml:1]: it doesnt' work with yarn 2+

File: .github/workflows/cd.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
+name: Release
+on:
+  workflow_run:
+    workflows: [ ci ]
+    branches: [ master ]
+    types:
+      - completed
```
Comment (mpetrunic) [inline .github/workflows/cd.yaml:7]: this doesn't check if ci failed: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-a-workflow-based-on-the-conclusion-of-another-workflow

**Review by mpetrunic — CHANGES_REQUESTED** [review]

File: .github/.dependabot.yaml (hunk 3 above)
Comment (BeroBurny) [reply .github/.dependabot.yaml:1]: for dependa bot, I followed instructions from the handbook - https://engineering-handbook.onrender.com/development/tech-stack/typescript#dependabot

Comment (mpetrunic) [reply .github/.dependabot.yaml:1]: thats for yarn 1. https://github.com/ChainSafe/engineering-handbook/pull/28/files#diff-65d1500c609e6708a8709aedda33ce9f13585733e71bd56188bd9f9dda8eefe9R125

**Comment by BeroBurny** [comment]
Will reopen PR after implementing renovate

File: .github/workflows/cd.yaml [hunk 5]
```diff
@@ -2,15 +2,14 @@ name: Release
 on:
   workflow_run:
     workflows: [ ci ]
-    branches: [ master ]
```
Comment (mpetrunic) [inline .github/workflows/cd.yaml]: why remove this?

File: packages/snap/src/index.ts [hunk 6]
```diff
@@ -1 +1,93 @@
...
+import { signPayloadJSON, signPayloadRaw } from "./rpc/substrate/sign";
+import { generateTransactionPayload } from "./rpc/generateTransactionPayload";
+import { send } from "./rpc/send";
+
+declare let wallet: Wallet;
+
+const apiDependentMethods = [
+  "getBlock", "getBalance", "getChainHead", "signPayloadJSON", "signPayloadRaw", "generateTransactionPayload", "send"
+];
+// eslint-disable-next-line
```
Comment (mpetrunic) [inline packages/snap/src/index.ts:20]: can you disable specific rule instead of everything.

File: packages/snap/src/index.ts [hunk 7]
```diff
@@ -1 +1,93 @@
...
+module.exports.onRpcRequest = (async ({ origin, request }: { origin: string, request: any }) => {
```
Comment (mpetrunic) [inline packages/snap/src/index.ts:21]:
```suggestion
module.exports.onRpcRequest = (async ({ origin, request }: { origin: string, request: unknown }) => {
```

File: .github/workflows/cd.yaml [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+  workflow_run:
+    workflows: [ ci ]
+    types:
+      - completed
+
+jobs:
+  maybe-release:
+    name: release
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push' && github.ref == 'refs/heads/master' && github.event.workflow_run.conclusion == 'success'
```
Comment (mpetrunic) [inline .github/workflows/cd.yaml:12]:
```suggestion
    if: github.event.workflow_run.conclusion == 'success'
```
`github.event_name == 'push'` - this won't work as event is workflow_run.
`github.ref == 'refs/heads/master'` - I left other comment

File: .github/workflows/cd.yaml [hunk 9]
```diff
@@ -0,0 +1,51 @@
+name: Release
+on:
+  workflow_run:
+    workflows: [ ci ]
```
Comment (mpetrunic) [inline .github/workflows/cd.yaml:4]:
```suggestion
    workflows: [ ci ]
    branches: [master]
```

**Review by mpetrunic — CHANGES_REQUESTED** [review]

**Review by mpetrunic — APPROVED** [review]

**Pull request merged**